### PR TITLE
Join classroom component

### PR DESCRIPTION
**Pull request by Erin007**
On the student homepage, students will have easy access to joining a section.

If the student is already enrolled in at least one section, they will see: 
join section - already enrolled in one

If the student is not yet enrolled in a section, they will see: 
join section - not enrolled yet

NOTE: This is just the UI for this component. It isn't yet being used anywhere.  The functionality for actually joining a session will come in a subsequent PR.

### CODE REVIEW

**Comment by Erin007** [comment]
cc: @tanyaparker

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -269,6 +269,10 @@
   "ignore": "Ignore",
   "infinity": "Infinity",
   "inspireStudents": "Inspire students",
+  "joinASection": "Join a Section",
```
Comment (breville) [inline apps/i18n/common/en_us.json:272]: I'm always torn whether the key names should match the literal string, so it's pretty readable in code as it shows, or whether they should match the purpose, in which case it's easier to change the content without modifying the key as well.  This seems fine, but I guess if we did the latter, we could name this key something like joinFirstSection.

File: apps/src/templates/studioHomepages/JoinSection.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import color from '@cdo/apps/util/color';
+import i18n from "@cdo/locale";
+import ProgressButton from '@cdo/apps/templates/progress/ProgressButton';
```
Comment (breville) [inline apps/src/templates/studioHomepages/JoinSection.jsx:4]: Should we rename this button at some point?

File: apps/src/templates/studioHomepages/JoinSection.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  mainDashed: {
+    borderWidth: 5,
+    borderStyle: 'dashed',
+    borderColor: color.border_gray,
+  },
+  heading: {
+    fontFamily: '"Gotham 4r", sans-serif',
+    fontSize: 16,
+    fontWeight: 'bold',
+    letterSpacing: -0.2,
```
Comment (breville) [inline apps/src/templates/studioHomepages/JoinSection.jsx:26]: ooh, what is this for?

File: apps/src/templates/studioHomepages/JoinSection.jsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    float: 'left',
+    marginTop: 15,
+    marginLeft: 20,
+    marginRight: 25
+  },
+};
+
+const JoinSection = React.createClass({
+  propTypes: {
+    enrolledInASection: React.PropTypes.boolean,
```
Comment (breville) [inline apps/src/templates/studioHomepages/JoinSection.jsx]: do we need A?

File: apps/src/templates/studioHomepages/JoinSection.jsx (hunk 2 above)
Comment (Erin007) [reply apps/src/templates/studioHomepages/JoinSection.jsx:4]: Yes. I think @Bjvanminnen is planning on making this change soon.

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (Erin007) [reply apps/i18n/common/en_us.json:272]: I've been trying to match the key name to the string if it's something short and likely reusable, such as "View course" and more closely matching purpose if the string is long/likely to only be used in that specific place. In this case, the user might not be joining their _first_ section so it seems okay to keep as is here.

File: apps/src/templates/studioHomepages/JoinSection.jsx (hunk 3 above)
Comment (Erin007) [reply apps/src/templates/studioHomepages/JoinSection.jsx:26]: It was in the original toast specs from Mike - I think it just compresses the text a little.

**Comment by Bjvanminnen** [comment]
Why do we need this UI to differentiate between the case where the student is/isn't in a section? It seems a little confusing to me, since we use that dashed line elsewhere to have different meaning, and here it doesn't really mean anything to the student.

**Comment by Erin007** [comment]
I think the reasoning is that the dashed border is more attention-grabbing.  The other "set up" components also have dashed borders; for example, if you don't have any courses yet, you see a big component with dashed borders inviting you to find a course. What does the dashed border mean elsewhere?

**Comment by Bjvanminnen** [comment]
We just started using it to mean whether a section is locked/hidden, though that will only be seen by teachers, so maybe nbd.

**Comment by Erin007** [comment]
@markabarrett This is the dashed line discussion Brendan mentioned on slack.  We've been using thick dashed line to draw attention to components where it's the user's first time doing something - joining a section, setting up a section, adding a course.  Does that conflict with Brent's use of dashed lines to indicate locked/hidden? They feel separate to me.

**Comment by markabarrett** [comment]
This is a really good question. I suggested other ways of indicating hidden, thick dashed outline is what I was told we were going with due to the time/difficulty of implementing any of the other affordances. I think it's not ideal but let's run with it for now and see if it is confusing or not. My hope is that it won't conflict as both states kind of indicate a place holder.

**Comment by markabarrett** [comment]
Oh, whoops, I see. I made a mistake. If all you are doing is calling attention, then I would not use the dashed outline. You could use a thick full outline, a notification badge, a tool tip, or a vignette. Did you add the dotted outline because users were not noticing the window? If that is not the problem, then I think you could have a standard outline.

**Pull request merged**